src/main/java/com/google/sps/servlets/ForumServlet.java

```diff
@@ -0,0 +1,120 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.gson.Gson;
+import com.google.sps.data.ForumElement;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles forum data (questions, comments, and replies) */
+@WebServlet("/forum")
+public class ForumServlet extends HttpServlet {
+
+  /** Get request to the Datastore and responds with the forum elements with the parentId input */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long parentId = Long.parseLong(request.getParameter("parentId"));
```

Comment:
parentId is the question itself right?

Reply:
Yeah that is a little confusing I will change that to be the id itself.  Thank you :)

Reply:
Updated the name to be id instead. Changed that also for doPost and for some functions in forum-script.js.  Much cleaner.